server/chat-plugins/mafia.ts

```diff
@@ -2220,6 +2263,29 @@ export const commands: ChatCommands = {
 		},
 		revealhelp: [`/mafia reveal [on|off] - Sets if roles reveal on death or not. Requires host % @ # &`],
 
+		takeidles(target, room, user) {
+			const args = target.split(',');
+			let targetRoom = Rooms.get(args[0]);
+			if (!targetRoom || targetRoom.type !== 'chat' || !targetRoom.users[user.id]) {
+				if (!room || room.type !== 'chat') return this.errorReply(`This command is only meant to be used in chat rooms.`);
+				targetRoom = room;
+			} else {
+				args.shift();
+			}
+			const game = targetRoom.getGame(MafiaTracker);
+			if (!game) return user.sendTo(targetRoom, `|error|There is no game of mafia running in this room.`);
+			if (game.hostid !== user.id && !game.cohosts.includes(user.id)) this.checkCan('mute', null, targetRoom);
+			const action = toID(args.join(''));
+			if (!['on', 'off'].includes(action)) return this.parse('/help mafia takeidles');
+			if ((action === 'off' && !game.takeIdles) || (action === 'on' && game.takeIdles)) {
+				return user.sendTo(targetRoom, `|error|Actions and idles are already ${game.takeIdles ? '' : 'not '}being accepted.`);
```

Comment:
Honestly, in general it seems like it would be cleaner to just set `this.room` to `targetRoom` here, so then you can just use the standard `this.errorReply` here. Am i missing something?

Reply:
This is the format that most commands for mafia use currently so I thought it'd be better to do it that way for now for consistency's sake, but I can go through the script and change all of them to `this.errorReply` in then next few days.

Reply:
(agree on this, good change overall, but out of scope for this PR)